src/ape/managers/query.py

```diff
@@ -1,13 +1,66 @@
-from typing import Dict, Optional
+from functools import partial
+from typing import Any, Dict, Optional
 
 import pandas as pd
+from pydantic import BaseModel
 
 from ape.api import QueryAPI, QueryType
+from ape.api.query import BlockQuery, _BaseQuery
 from ape.exceptions import QueryEngineError
 from ape.plugins import clean_plugin_name
 from ape.utils import ManagerAccessMixin, cached_property
 
 
+def get_columns_from_item(query: _BaseQuery, item: BaseModel) -> Dict[str, Any]:
+    return {k: v for k, v in item.dict().items() if k in query.columns}
+
+
+class DefaultQueryProvider(QueryAPI):
+    """
+    Default implementation of the ape.api.query.QueryAPI
+    Allows for the query of blockchain data
+
+    Args:
```

Comment:
This doc is incorrect, at class level you don't have args or returns

Reply:
So add the docs on the QueryAPI?

Reply:
Sorry, see https://github.com/ApeWorX/ape/pull/535#discussion_r820127534